minigunapp/views.py

```diff
@@ -0,0 +1,62 @@
+import json
+import yaml
+
+from django.http import HttpResponse, Http404, JsonResponse
+from django.shortcuts import render
+from cerberus import Validator
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import authentication, permissions
+
+from minigunapp.models import Email
+from minigunapp.serializers import EmailSerializer
+from minigunapp.tasks import send_email
+
+
+v = Validator()
+schema = yaml.load(open('minigunapp/schemas/email.yaml'))
+page_size = 10
+
+
+class EmailList(APIView):
+
+    def get(self, request, format=None):
+        """ Returns a single page of emails in Json format """
+
+        page = int(request.GET.get('page', 1))
+        page = page if page > 0 else 1
+        start = page_size * page
+        end = page_size * (page + 1)
+
+        emails = Email.objects[start:end]
+        emails = [EmailSerializer(email).data for email in emails]
```

Comment:
Ok I think I see what you're asking. This route does return a list but it could return a json object with the following sort of format:

```
{
    "field1": "something",
    "field2": "something",
    "emails": [
       ...
    ]
}
```

Versioning the API is a nice way of dealing with schema changes. It's typical to root your routes at /v1/, /v2/ etc. You could also publish your schema in a consumable way, as with xsd files, for a client program to consume.